src/xmipp/libraries/reconstruction/volumeset_align.cpp

```diff
@@ -126,8 +129,16 @@ void ProgVolumeSetAlign::computeFitness(){
 
 	int err;
 
-	runSystem("xmipp_volume_align",formatString("--i1 %s --i2 %s --frm %f %d %d %d --store %s -v 0 ",
-			Volume1,Volume2,this->frm_freq, this->frm_shift, this->tilt0, this->tiltF, shifts_angles));
+	String fnMask;
+	if (this->mask_option){
+		fnMask = this->mask_fn;
+		runSystem("xmipp_volume_align",formatString("--i1 %s --i2 %s --frm %f %d %d %d --store %s -v 0 --mask %s",
+			Volume1,Volume2,this->frm_freq, this->frm_shift, this->tilt0, this->tiltF, shifts_angles, fnMask));}
+	else{
+		runSystem("xmipp_volume_align",formatString("--i1 %s --i2 %s --frm %f %d %d %d --store %s -v 0 ",
```

Comment:
extract arguments into a separate string and if necessary, append the mask argument